### PR TITLE
Fixed issue 29920 -  "Item in Cart" not pluralized correctly

### LINKED ISSUE #29920: "Item in Cart" not pluralized correctly
### Preconditions (*)
Magento 2.3.5
Sample Data installed

### Steps to reproduce (*)
Add a single product to the cart in a quantity of 3. (Or, any quantity greater than one.)

### Expected result (*)
In the minicart the text "3 Items in Cart" appears.
In the checkout order summary the text "3 Items in Cart" appears.

### Actual result (*)
In the minicart the text "3 Item in Cart" appears.
In the checkout order summary the text "3 Item in Cart" appears.

### Additional Information (*)
Seem problem happen when use config Display Cart Summary: Display item quantities <=Default option
Admin > Configuration > Sales > Checkout > My Cart Link

Test case 1: Add only one product and update qty to above a least 2 => NOT WORK
Test case 2: Add 2 products and update qty (optional) => WORK

The other option "Display number of items in cart" seem work

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [X] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

### PR DESCRIPTION
**Pull request by saphaljha**
### Description (*)
When adding more then 1 qty of an item in cart and check checkout page order summary section text 'Items in Cart' is pluralized correctly or not

### Fixed Issues (if relevant)

1. Fixes magento/magento2#29920

### Manual testing scenarios (*)

1. Manually Tested
2. Added MFTF

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by saphaljha** [comment]
@magento run all tests

**Review by rogyar — CHANGES_REQUESTED** [review]
Hi @saphaljha. Thank you for your contribution.
Please, fill out the PR description according to the template (including steps to reproduce and other details).
Also, I would ask you to cover your fix using a simple MFTF test. Thank you!

**Comment by saphaljha** [comment]
Hello @rogyar,

Thank you for review, I will update it as suggested.

Thank you

**Comment by saphaljha** [comment]
@magento run all tests

**Comment by saphaljha** [comment]
@magento run Functional Tests B2B

File: app/code/Magento/Checkout/Test/Mftf/Test/StorefrontGuestCheckoutTest/StorefrontCheckoutSummaryItemsInCartLabelPluralized.xml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
+    <test name="StorefrontCheckoutSummaryItemsInCartLabelPluralized">
```
Comment (rogyar) [inline app/code/Magento/Checkout/Test/Mftf/Test/StorefrontGuestCheckoutTest/StorefrontCheckoutSummaryItemsInCartLabelPluralized.xml]: Hi @saphaljha awesome work! Thank you. One minor thing, please, add the "...Test" suffix to the test name and the file name correspondingly
```suggestion
    <test name="StorefrontCheckoutSummaryItemsInCartLabelPluralizedTest">
```

**Review by rogyar — CHANGES_REQUESTED** [review]

File: app/code/Magento/Checkout/Test/Mftf/Test/StorefrontGuestCheckoutTest/StorefrontCheckoutSummaryItemsInCartLabelPluralized.xml (hunk 1 above)
Comment (saphaljha) [reply app/code/Magento/Checkout/Test/Mftf/Test/StorefrontGuestCheckoutTest/StorefrontCheckoutSummaryItemsInCartLabelPluralized.xml]: @rogyar Thank you :) , I have updated commit as per your suggestions, thank you for review.

**Comment by saphaljha** [comment]
@magento run all tests

**Comment by saphaljha** [comment]
@magento run Functional Tests CE

**Review by rogyar — DISMISSED** [review]

**Review by engcom-Alfa — CHANGES_REQUESTED** [review]
Hi @saphaljha .

Unfortunately, looks like the changes don't solve the issue #29920  completely.

### Case 1.

Add a single product to the cart in a quantity of 3. (Or, any quantity greater than one)

**Actual Result:**

:heavy_check_mark: In the checkout order summary the text "3 Items in Cart" appears.

Screenshot from 2020-09-21 15-27-26


:heavy_multiplication_x: In the mini cart the text "3 Item in Cart" appears.




### Case 2.

1. Add a single product to the cart in a quantity of 1;
2. In the mini cart change quantity from 1 to 2 and click on the **Update** button;

**Actual Result:** :heavy_multiplication_x: In the mini cart the text "2 Item in Cart" appears.

Peek 2020-09-21 15-20

@saphaljha Could you take a look?
Thanks!

**Comment by mrtuvn** [comment]
@engcom-Alfa this in my opinion seem not a bug. Default magento restrict maximum products item line (default: 10) added to minicart. When you add more than 10 unique item magento will show 10 of  max total items added
Items in Cart indicate number of product items  line added in cart

**Comment by saphaljha** [comment]
Hello @engcom-Alfa , I am working on these 2 cases.

**Comment by sidolov** [comment]
@saphaljha I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Comment by saphaljha** [comment]
@sidolov @rogyar ,

I have updated as per suggestions.

@sidolov  Thank you for reopen PR.

**Comment by saphaljha** [comment]
@magento run all tests

**Comment by saphaljha** [comment]
@magento run Functional Tests CE
@magento run Functional Tests EE,
@magento run Functional Tests B2B
@magento run Integration Tests

**Comment by rogyar** [comment]
Hi @saphaljha. Please, adjust the existing functional tests to match the new behavior.
Thanks.

**Comment by saphaljha** [comment]
Hello @rogyar ,

You mean to say I need to cover MFTF as per cases like for minicart mentioned above by @engcom-Alfa ?
Because I have covered it in MFTF but flow is different.

Thanks to guide me in that.

**Comment by rogyar** [comment]
Hi @saphaljha. I mean, we need to adjust assertions for some of the existing tests since they (tests) still expect the old values. Please, check the failing tests for details. 

Thank you.

**Comment by saphaljha** [comment]
@magento run all tests

**Comment by saphaljha** [comment]
@magento run all tests

**Comment by saphaljha** [comment]
@magento run Functional Tests B2B
@magento run WebAPI Tests

**Comment by saphaljha** [comment]
@magento run WebAPI Tests

**Review by rogyar — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @rogyar, thank you for the review.
[ENGCOM-8447](https://jira.corp.magento.com/browse/ENGCOM-8447) has been created to process this Pull Request

**Comment by engcom-Bravo** [comment]
:heavy_check_mark: QA Passed

### Manual testing scenario

1. Add a single product to the cart in a quantity of 3. (Or, any quantity greater than one.)

*Actual Result:* on Magento 2.4-develop
In the minicart the text "2 Item in Cart" appears.
Before PR
In the checkout order summary the text "3 Item in Cart" appears.
Summary before PR

**After applying changes provided in this PR**
In the minicart the text "2 Items in Cart" appears.
After PR
In the checkout order summary the text "3 Items in Cart" appears.
Summary after PR
Also rechecked with one single item in Cart **1 Item in Cart** is displayed

**Pull request merged**